### DISCUSSION TITLE
Strict convertion to int in some transform

### DISCUSSION THREAD

**Discussion by rcremese** [Project-MONAI/MONAI#7170] [General]
Hello, 
I don't want to polute the issue tracker with this but won't it be possible to be more permissive types in transforms like [Resize](https://docs.monai.io/en/latest/transforms.html#resize) ? 
I fact I realized an error is raised when using the "longest" size_mode.
`Resize(new_dim, size_mode="longest", mode="nearest")`
If new_dim is an instance of numpy.int64 or  torch.int32 it raise the following exception : 
```python
Exception has occurred: RuntimeError
applying transform <monai.transforms.spatial.array.Resize object at 0x7f579c129450>
ValueError: spatial_size must be an int number if size_mode is 'longest'.
```
I understand why but can't it convert value to an `int` first and raise the error if the type is different ?

**Comment by KumoLiu** [comment]
Hi @rcremese, since the `spatial_size` is entered directly by the user, the initial idea may be that the user can make sure that they are entering an int, rather than changing the user's input internally, but perhaps we can add a warning here to make it more flexible.
cc @wyli

**Reply by wyli** [reply]
Agreed, probably we can remove the type check or change it to a warning..
